Library/Formula/xsd.rb

```diff
@@ -0,0 +1,49 @@
+class Xsd < Formula
+  desc "XML Data Binding for C++"
+  homepage "http://www.codesynthesis.com/products/xsd/"
+  url "http://www.codesynthesis.com/download/xsd/4.0/macosx/i686/xsd-4.0.0-i686-macosx.tar.bz2"
+  sha256 "56571e446ae49935611cbf46484a50c1a0afd3bb7987be728b8dcdd3de34b897"
+
+  depends_on "xerces-c"
+
+  def install
+    bin.install "bin/xsd"
+    include.install "libxsd/xsd"
+  end
+
+  test do
+    schema = (testpath/"meaningoflife.xsd")
+    schema.write <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified"
+               targetNamespace="http://brew.sh/XSDTest" xmlns="http://brew.sh/XSDTest">
+        <xs:element name="MeaningOfLife" type="xs:positiveInteger"/>
+    </xs:schema>
+    EOS
+    instance = (testpath/"meaningoflife.xml")
+    instance.write <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <MeaningOfLife xmlns="http://brew.sh/XSDTest" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://brew.sh/XSDTest meaningoflife.xsd">
+        42
+    </MeaningOfLife>
+    EOS
+    xsdtest = (testpath/"xsdtest.cxx")
+    xsdtest.write <<-EOS.undent
+    #include <cassert>
+    #include "meaningoflife.hxx"
+    int main (int argc, char *argv[]) {
+        assert(2==argc);
+        std::auto_ptr< ::xml_schema::positive_integer> x = XSDTest::MeaningOfLife(argv[1]);
+        assert(42==*x);
+        return 0;
+    }
+    EOS
+    system "#{bin}/xsd", "cxx-tree", schema
+    assert File.exist? (testpath/"meaningoflife.hxx")
+    assert File.exist? (testpath/"meaningoflife.cxx")
+    system "c++", "-o", "xsdtest", "xsdtest.cxx", "meaningoflife.cxx", "-lxerces-c"
+    assert File.exist? (testpath/"xsdtest")
+    system (testpath/"xsdtest"), instance
+  end
```

Comment:
On 2 June 2015 at 11:15, Mike McQuaid <notifications@github.com> wrote:

> In Library/Formula/xsd.rb
> <https://github.com/Homebrew/homebrew/pull/40274#discussion_r31504699>:
>
> > +    #include <cassert>
> > +    #include "meaningoflife.hxx"
> > +    int main (int argc, char *argv[]) {
> > +        assert(2==argc);
> > +        std::auto_ptr< ::xml_schema::positive_integer> x = XSDTest::MeaningOfLife(argv[1]);
> > +        assert(42==*x);
> > +        return 0;
> > +    }
> > +    EOS
> > +    system "#{bin}/xsd", "cxx-tree", schema
> > +    assert File.exist? (testpath/"meaningoflife.hxx")
> > +    assert File.exist? (testpath/"meaningoflife.cxx")
> > +    system "c++", "-o", "xsdtest", "xsdtest.cxx", "meaningoflife.cxx", "-lxerces-c"
> > +    assert File.exist? (testpath/"xsdtest")
> > +    system (testpath/"xsdtest"), instance
> > +  end
>
> The parens here (and most others in this method) aren't needed.
>

Ok, I will fix this later today.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/40274/files#r31504699>.
>